### ISSUE TITLE
Expand `skore.train_test_split` with several warnings

### ISSUE THREAD

**Issue by augustebaum** [probabl-ai/skore#684]
Implements #492:

- [x] W1: high class imbalance (#690)
- [x] W1 bis: a class has fewer than 100 examples (#772)
- [x] W2: stratify is set (#780)
- [x] W3: random_state is unset (#781)
- [x] W4: datetime column (#809)
- [x] W6: shuffle is true (#791)

Follows #683
